Documentation/compilation.rst

```diff
@@ -52,78 +81,88 @@ Installing HDF5
 
 .. code-block:: bash
 
-  wget https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.8/hdf5-1.8.11/src/hdf5-1.8.11.tar.bz2
-  tar -xaf hdf5-1.8.11.tar.bz2
-  cd hdf5-1.8.11
-  CC=mpiicc FC=mpiifort ./configure --enable-parallel --prefix=$HOME --with-zlib --disable-shared --enable-fortran 
-  #NOTE: This is with Intel MPI, if you have to use a different MPI, please adjust mpiicc and mpiifort accordingly to the scripts that call icc and ifort!
-  #MPICH note: CC=mpicc FC=mpif90 ./configure --enable-parallel --prefix=$HOME --with-zlib --disable-shared --enable-fortran
-  make
+  git clone https://github.com/mortenpi/hdf5.git
+  cd hdf5
+  git checkout 8a275ab7831002f3e  # take a stable version (optional)
+  ./configure --prefix=$S3_HOME --enable-parallel --with-zlib --disable-shared --enable-fortran
+  make -j4
   make install
   cd ..
 
+
 Installing netCDF
 -----------------
 
 .. code-block:: bash
 
   wget ftp://ftp.unidata.ucar.edu/pub/netcdf/netcdf-4.4.1.1.tar.gz
-  tar -xaf netcdf-4.3.0.tar.gz
-  cd netcdf-4.3.0
-  CPPFLAGS=-I$HOME/include LDFLAGS=-L$HOME/lib CC=mpiicc ./configure --enable-shared=no --prefix=$HOME 
-  #MPICH note: CPPFLAGS=-I$HOME/include LDFLAGS=-L$HOME/lib CC=mpicc ./configure --enable-shared=no --prefix=$HOME 
-  #NOTE: Check for this line to make sure netCDF is build with parallel I/O: 
-  #"checking whether parallel I/O features are to be included... yes" This line comes at the very end (last 50 lines of configure run)!
-  make
+  tar -xaf ./netcdf-*.tar.gz
+  cd netcdf-4.4.1.1
+  CPPFLAGS="-I$S3_HOME/include -fPIC" ./configure --prefix=$S3_HOME --enable-shared=no
+  make -j4
+  make check # to check correctness of installation (optional)
   make install
   cd ..
 
 
+
 Installing Libxsmm
 ------------------
 
 .. code-block:: bash
 
-   git clone https://github.com/hfp/libxsmm
-   cd libxsmm
-   make generator
-   #add libxsmm/bin to path
+  git clone https://github.com/hfp/libxsmm
+  cd libxsmm
+  git checkout b6de187f832a723295a  # take a stable version (optional) 
+  make generator
+  cp ./bin/libxsmm_gemm_generator $S3_HOME/bin/
+  cd ..
 
 
 Installing Metis (Optional:PUML mesh format)
 --------------------------------------------
 
 .. code-block:: bash
 
-   tar -xvf metis-5.1.0.tar.gz
-   cd metis-5.1.0
-   #edit ./include/metis.h IDXTYPEWIDTH to be 64 (default is 32).
-   make config cc=mpiicc cxx=mpiicpc prefix=$HOME #(must confirm compiler!!!)
-   make install
-   cd ..
+  wget http://glaros.dtc.umn.edu/gkhome/fetch/sw/metis/metis-5.1.0.tar.gz
+  tar -xvf ./metis-5.1.0.tar.gz
+  cd metis-5.1.0
```

Comment:
why did you skip editing the metis.h file? Is is not necessary?

Reply:
I checked installation of metis on several machines. You don't need to edit the header at all

Reply:
As far as I've understood this, it is not required for compilation but rather for very large graphs (where the 32bit length becomes restrictive).
Probably @uphoffc knows more about this.

Reply:
Yes the 64 bits are required for large meshes. Might make sense to mark it as optional and to provide a patch in the documentation, such that manual editing is no longer required.
Regards,
Carsten